sdk/src/logs/logger.cc

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+Logger::Logger(std::shared_ptr<LogProcessor> processor) noexcept : processor_{processor} {}
+
+void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
+{
+  // TODO: Sampler
+
+  // TODO: before calling IsEnabled(), should check whether this record's severity is set.
+  // If it is not set, set it to the default (kInfo).
+
+  // If this log's severity is less than the minimum severity, do nothing
+  if (!IsEnabled(record.severity))
+  {
+    return;
+  }
+
+  // If this logger does not have a processor, no need to create a log record
+  auto processor = GetProcessor();
+  if (processor == nullptr)
+  {
+    return;
+  }
+
+  /**
+   * Convert the LogRecord to the heap first before sending to processor.
+   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
+   * converting record a heap variable can be removed
+   */
+  auto record_pointer =
+      std::unique_ptr<opentelemetry::logs::LogRecord>(new opentelemetry::logs::LogRecord(record));
+
+  // Inject timestamp
+  // TODO: add a flag in the API to check if timestamp is set by user already or not, then inject
```

Comment:
Are these TODOs tracked anywhere for follow up outside of comments?

Reply:
Most of the TODOs here, except the sampler, will be addressed by us (we hope) in a future PR. Since some of the PRs require changes to the Logging API, we were planning to PR a Logging API Update first then address some of the TODOs here

Reply:
Yea most of these TODOs are from the original [Logging prototype issue](https://github.com/open-telemetry/opentelemetry-cpp/issues/337).